src/widget.js

```diff
@@ -77,14 +77,14 @@ var Widget = Base.$extend({
         // wEvents
         this._registerWEvents(["show", "hide"]);
 
+        // Name must be set before other properties (e.g. needed when setting children)
+        if (!this._name) {
+            this.name = params && params.name ? params.name : "widget-" + uuid.v4();
```

Comment:
if params.name is set, it will be written two times, but I guess it cannot be helped.

Reply:
I can remove `name` from `params` but it is a bit of a hack no ?

Reply:
Yeah, that sound like a hack. I guess it's not that bad.